torch/_inductor/codegen/cpp.py

```diff
@@ -2062,9 +2059,36 @@ def add_to_dtype(sub_graph: torch.fx.Graph):
                     pass
 
             def eliminate_to_dtype(sub_graph: torch.fx.Graph):
-                # TODO(Eikan) Remove redundant to_dtype like load_bf16 + to_fp32 + to_bf16 + store_bf16
-                # => load_bf16 + store_bf16
-                pass
+                def _eliminate_redundant_to_node(sub_graph: torch.fx.Graph):
+                    # Eliminate the redudant to_dtype node. Let's consider a pattern as follows:
+                    #   graph():
+                    #     %to_dtype1 = call_method[target=to_dtype](args = (%ops, %input, torch.float), kwargs = {})
+                    #     %to_dtype2 = call_method[target=to_dtype](args = (%ops, %to_dtype1, torch.float), kwargs = {})
+                    # Regarding the first to_dtype, it is redudant because the second to_type also converts to the
+                    # torch.float. Hence, we remove the first to_type
+                    def _used_by_to(to_node: torch.fx.Node):
+                        return all(usr.target == "to_dtype" for usr in to_node.users)
+
+                    all_to_nodes = [
+                        node for node in sub_graph.nodes if node.target == "to_dtype"
+                    ]
+                    all_to_nodes_and_users = [
+                        {node: node.users} for node in all_to_nodes if _used_by_to(node)
+                    ]
+                    for node_users in all_to_nodes_and_users:
+                        for node, users in node_users.items():
+                            if all(usr.args[-1] == node.args[-1] for usr in users):
+                                val_node = node.all_input_nodes[-1]
+                                node.replace_all_uses_with(val_node)
+                                sub_graph.erase_node(node)
+
+                    sub_graph.lint()
+
+                def _eliminate_to_node_mem_copy(sub_grah: torch.fx.Graph):
```

Comment:
What does this function do?

Reply:
`load_bf16 + to_fp32 + to_bf16 + store_bf16` => `load_bf16 + store_bf16`